psi4/src/psi4/liboptions/liboptions.cc

```diff
@@ -327,6 +327,8 @@ Data::Data(DataType* t) { ptr_ = std::shared_ptr<DataType>(t); }
 
 Data::Data(const Data& copy) { ptr_ = copy.ptr_; }
 
+Data& Data::operator=(const Data& copy) { ptr_ = copy.ptr_; return *this; }  // ? ptr_.release();
```

Comment:
@jturney there doesn't look to be any allocating going on for this class. would a `ptr_->reset();` be good in place of the `release()`?

Reply:
Yeah, `reset()` makes sense.

Reply:
@jturney, the code hated the `reset()` (compiled but segfaulted; backtrace below), so I took it back out. Ok, or should I investigate further?

```
(gdb) bt
#0  0x00007ffff01ae6e2 in psi::Data::reset() () from /home/psilocaluser/gits/hrw-patch/objdir4/stage/lib/psi4/core.cpython-37m-x86_64-linux-gnu.so
#1  0x00007ffff01ae7f0 in psi::Data::operator=(psi::Data const&) () from /home/psilocaluser/gits/hrw-patch/objdir4/stage/lib/psi4/core.cpython-37m-x86_64-linux-gnu.so
#2  0x00007ffff01b41d7 in psi::Options::add(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, psi::DataType*) ()
   from /home/psilocaluser/gits/hrw-patch/objdir4/stage/lib/psi4/core.cpython-37m-x86_64-linux-gnu.so
#3  0x00007fffef2fb29a in psi::read_options(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, psi::Options&, bool) ()
   from /home/psilocaluser/gits/hrw-patch/objdir4/stage/lib/psi4/core.cpython-37m-x86_64-linux-gnu.so
#4  0x00007fffef09f3c6 in psi4_python_module_initialize() () from /home/psilocaluser/gits/hrw-patch/objdir4/stage/lib/psi4/core.cpython-37m-x86_64-linux-gnu.so
```